nav2_bringup/launch/system_test.yaml

```diff
@@ -0,0 +1,10 @@
+map_server:
```

Comment:
Would it be possible to not nest the yaml file with map_server: ros__parameters? this is for the map server and those 2 namespaces seems redundant given that this yaml file is for the map it matches to, not for the node. This also makes it so you couldn't use your maps generated with ROS1 or any other program work out of the box in ROS2.

Reply:
@SteveMacenski  I don't think so. We can keep it ROS1 compatible, but that means the map server node would have to do its own YAML parsing and not take advantage of the standard ROS2 YAML parameter parsing. This introduces a dependency of the map server on a YAML library. 

Also, we are considering a UI that queries the various node parameters and then presents a generic interface for parameter tuning. If the file is kept ROS1 compatible and the node does the parsing of the YAML file, these wouldn't be node parameters and wouldn't show up in the UI. I don't know if this is important or not. 

Finally, the parameters in the file are almost all for the conversion process from bitmap to OccupancyGrid. That is, they are parameters for the OccGridLoader and not really metadata for the map itself. There is one parameter, resolution, that could be considered metadata, but the others, like occupied_thresh and free_thresh are about the conversion and not specific to the map.

Reply:
All good points. 

For the UI, these values aren't something to be "tuned" like a controller or timing, so I wouldn't think the UI would be helpful. Having to include your own parser is a pain point. For existing users with large libraries of maps, this would be fairly annoying to have to convert them all. Is there a middle ground you can think of to allow support for ROS1 configs without including the yaml parsing code?

Reply:
@SteveMacenski I've updated the code to implement what we discussed a the recent Navigation WG meeting. Please have a look.